package.json

```diff
@@ -25,12 +25,8 @@
     },
     "projectConfigFile": {
       "type": "array",
-      "default": [
-        "tox.ini",
-        ".flake8",
-        "setup.cfg"
-      ],
-      "description": "flake config file relative path from project (e.g. tox.ini or .flake8rc)",
+      "default": [],
```

Comment:
This setting is used to determine whether or not a configuration file exists for the project, which then controls whether the overrides are taken into account or not. So making it blank like this entirely breaks that functionality.

A better way to handle this would be to have this setting be an _addition_ to the list of configuration files, with the default list `flake8` searches for stored in the code. Use the merged list to see if there is a config file at all, and use the custom list to determine if `--config` needs to be specified.

Reply:
Doesn't Flake8 already find and use the appropriated configuration file? As it is now `--config` always override the default behavior due to the default values. Check the discussion on #554

Reply:
With the current state of this PR [this condition](https://github.com/AtomLinter/linter-flake8/blob/0f3d35965a0761c6cb229ebc7a1d3938215376a2/lib/main.js#L212) will virtually always evaluate to `false`, meaning all of the settings like [these](https://github.com/AtomLinter/linter-flake8/blob/0f3d35965a0761c6cb229ebc7a1d3938215376a2/package.json#L38-L65) that add in override flags like `--max-complexity` will come into play, which would stomp all over any configuration file `flake8` uses.

The detection of the configuration file is what needs to be fixed _first_, then once the configuration file is being properly detected we can decide whether or not to force it with `--config`. I was proposing a super basic way of if one of the default config files is found (`.flake8`, `tox.ini`...) to not pass `--config` and _also_ not pass things like `--max-complexity`. If the user specifies a custom file in the settings and it is detected then that would always be sent in via `--config`.

An extension to that, which sounds like it is necessary for files like `tox.ini` is to determine if they have a `flake8` section, and if not then allow the overrides like `--max-complexity` through.

Reply:
I'm fine with keeping this PR on hold and I may take a stab at detecting default config files. Do you want a separate issue for that?

Reply:
Up to you, I'm fine with this being the "fix all the config things" PR 😛.